src/microbe_stage/Microbe.cs

```diff
@@ -904,10 +904,28 @@ public void AIThink(float delta, Random random, MicrobeAICommonData data)
     /// </returns>
     public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
     {
+        HashSet<(Compound Compound, float Range, float MinAmount, Color Colour)> colonyCompoundDetections;
```

Comment:
This variable doesn't actually always hold the "colony" detections, so the name should be something else to actually describe what it holds.

Reply:
I renamed it to `collectedUniqueCompoundDetections`.